tests/__init__.py

```diff
@@ -47,3 +47,9 @@ def rm_files(file_paths):
             elif os.path.isdir(file_path):
                 shutil.rmtree(file_path, ignore_errors=False, onerror=None)
 
+
+def validate_callback(mock, validate_func):
```

Comment:
Could we perhaps create a new instance of `Mocker` here and avoid spamming it in the code of the tests?

Reply:
I don't get the suggestion :thinking:

Reply:
We could pass `mocker` instead of `mock` as an argument and use `mocker.patch()` or `mocker.patch.object`.

Reply:
Do we need to create `m = mocker.Mock()` in the body of the test? Can't we create it inside this `validate_cb` instead, for each situation?

Reply:
Yes, because it is send to the Flow as the `on_done` callback and collects the callback data there.

Reply:
we could maybe offer this as a fixture that yields the mock and when finished passes the callback through this validation?

Reply:
This would need to be very well documented however